src/components/resource-selector/room-media/room-media-screens.js

```diff
@@ -132,37 +129,12 @@ function DocumentOrRoomMediaScreens({ storageLocationType, initialUrl, onSelect,
   };
 
   useEffect(() => {
-    const checkPreconditions = () => {
-      return storageLocation.type === STORAGE_LOCATION_TYPE.documentMedia
-        ? new Promise(resolve => {
-          if (!getCookie(uploadLiabilityCookieName)) {
-            confirmPublicUploadLiability(t, () => {
-              setSessionCookie(uploadLiabilityCookieName, 'true');
-              resolve(true);
-            }, () => resolve(false));
-          } else {
-            resolve(true);
-          }
-        })
-        : true;
-    };
-
-    const startUpload = async () => {
-      if (!uploadQueue.length) {
-        return;
-      }
-
-      const preMet = await checkPreconditions();
-      if (!preMet || !uploadQueue.length) {
-        return;
-      }
-
-      pushScreen(SCREEN.upload);
-    };
-
-    startUpload();
-
-  }, [uploadQueue, storageLocation.type, uploadLiabilityCookieName, t]);
+    if (!uploadQueue.length) {
+      return;
+    }
+
+    pushScreen(SCREEN.upload);
+  }, [uploadQueue, uploadLiabilityCookieName, t]);
```

Comment:
I think we don't need `uploadLiabilityCookieName` and `t` here anymore

Reply:
Indeed, cleaned it up.